### PR TITLE
fix: show username first and then password when credentials are listed

### PR DESCRIPTION
**Pull request by DhariniJeeva**
✅ Closes: https://hashicorp.atlassian.net/browse/ICU-15105

# Description

This PR moves up `username` field  

## Screenshots (if appropriate)

before:

Screenshot 2024-09-17 at 6 37 46 PM

after:
Screenshot 2024-09-17 at 6 31 53 PM

with a JSON blob: 
Screenshot 2024-09-18 at 6 22 21 PM

## Checklist

- [x] I have added before and after screenshots for UI changes
- ~[ ] I have added JSON response output for API changes~
- [x] I have added steps to reproduce and test for bug fixes in the description
- [x] I have commented on my code, particularly in hard-to-understand areas
- [x] My changes generate no new warnings
- [x] I have added tests that prove my fix is effective or that my feature works

### CODE REVIEW

**Comment by DhariniJeeva** [comment]
I created this against `main` since this does not impact any TS changes, but let me know if you think otherwise, happy to move this to llb

**Review by cameronperera — APPROVED** [review]
LGTM. Thanks for getting this updated!

File: ui/desktop/tests/integration/helpers/order-by-test.js [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+import { hbs } from 'ember-cli-htmlbars';
+
+module('Integration | Helper | order-by', function (hooks) {
+  setupRenderingTest(hooks);
+
+  test('it renders', async function (assert) {
+    this.set('inputValue', [
+      { key: 'username', value: 1 },
+      { key: 'password', value: 2 },
+      { key: 'none', value: 3 },
```
Comment (ZedLi) [inline ui/desktop/tests/integration/helpers/order-by-test.js:18]: What happens if you have other keys that are in between username and password? Does that order make sense?

File: ui/desktop/app/components/credentials-panel/index.hbs [hunk 2]
```diff
@@ -61,7 +62,7 @@
 
       <ul class='credential-secret'>
         {{#if credential.secrets}}
-          {{#each credential.secrets as |secret|}}
+          {{#each (order-by credential.secrets 'username') as |secret|}}
```
Comment (ZedLi) [inline ui/desktop/app/components/credentials-panel/index.hbs:65]: Shouldn't username and password be next to each other? How do we handle if a user has many different keys and username and password might be far apart from each other?

Should we make this more generic to handle any arbitrary sorting order?

Comment (DhariniJeeva) [reply ui/desktop/app/components/credentials-panel/index.hbs:65]: that's a valid point, looking into this

Comment (DhariniJeeva) [reply ui/desktop/app/components/credentials-panel/index.hbs]: changed the function logic to follow the custom order array we pass. So now, the objects will be ordered based on whatever custom order is passed to the function. Let me know what you think

File: ui/desktop/tests/integration/helpers/order-by-test.js [hunk 3]
```diff
@@ -0,0 +1,37 @@
...
+
+import { module, test } from 'qunit';
+import { setupRenderingTest } from 'ember-qunit';
+import { render } from '@ember/test-helpers';
+import { hbs } from 'ember-cli-htmlbars';
+
+module('Integration | Helper | order-by', function (hooks) {
+  setupRenderingTest(hooks);
+
+  test('it renders', async function (assert) {
```
Comment (ZedLi) [inline ui/desktop/tests/integration/helpers/order-by-test.js:14]: Could we add a few more test cases to show the different scenarios, e.g. preserves sort order if no keys are in the list, some keys are in the list, etc

**Review by ZedLi — APPROVED** [review]
Looks good!

File: ui/desktop/tests/integration/helpers/order-by-test.js (hunk 3 above)
Comment (DhariniJeeva) [reply ui/desktop/tests/integration/helpers/order-by-test.js:14]: good call, added!

File: ui/desktop/app/components/credentials-panel/index.js [hunk 4]
```diff
@@ -8,6 +8,8 @@ import { tracked } from '@glimmer/tracking';
 import { action } from '@ember/object';
 
 export default class CredentialsPanelIndexComponent extends Component {
+  customOrder = ['username', 'password'];
```
Comment (calcaide) [inline ui/desktop/app/components/credentials-panel/index.js:11]: I really like this! Amazing

**Review by calcaide — APPROVED** [review]
Terrific job!! Thank you for this helper and the "issue" it fixes.

**Pull request merged**